packages/svelte-query/src/createMutation.svelte.ts

```diff
@@ -24,48 +23,62 @@ export function createMutation<
   options: Accessor<CreateMutationOptions<TData, TError, TVariables, TContext>>,
   queryClient?: Accessor<QueryClient>,
 ): CreateMutationResult<TData, TError, TVariables, TContext> {
-  const client = useQueryClient(queryClient?.())
+  const client = $derived(useQueryClient(queryClient?.()))
 
-  const observer = $derived(
+  // svelte-ignore state_referenced_locally - intentional, initial value
+  let observer = $state(
+    // svelte-ignore state_referenced_locally - intentional, initial value
     new MutationObserver<TData, TError, TVariables, TContext>(
       client,
       options(),
     ),
   )
 
-  const mutate = $state<
-    CreateMutateFunction<TData, TError, TVariables, TContext>
-  >((variables, mutateOptions) => {
-    observer.mutate(variables, mutateOptions).catch(noop)
-  })
+  watchChanges(
+    () => client,
+    'pre',
+    () => {
+      observer = new MutationObserver(client, options())
+    },
+  )
 
   $effect.pre(() => {
     observer.setOptions(options())
   })
 
-  const result = $state(observer.getCurrentResult())
-
-  const unsubscribe = observer.subscribe((val) => {
-    notifyManager.batchCalls(() => {
-      Object.assign(result, val)
-    })()
+  const mutate = <CreateMutateFunction<TData, TError, TVariables, TContext>>((
+    variables,
+    mutateOptions,
+  ) => {
+    observer.mutate(variables, mutateOptions).catch(noop)
   })
 
-  onDestroy(() => {
-    unsubscribe()
+  let result = $derived(observer.getCurrentResult())
```

Comment:
@lachlancollins do you think we need to keep the Svelte peer dep pinned to as low as `5.7.0`, or could we bump that to at least `5.25.0` [so that assigning to `$derived` like this is supported](https://svelte.dev/docs/svelte/$derived#Overriding-derived-values)? This was part of `eslint --fix`, but if we need to stay on the lower version, I'll have to revert this back to a manually managed `$state`.

Reply:
I'm happy to bump the peer dependency version!

Reply:
@hmnd I've updated this on the target branch.

Reply:
It seems like this change(?) might have caused tests to start failing. It's difficult to know though, as the force pushes have erased the history.

Reply:
Thank you for getting this merged @lachlancollins! Sorry about the force pushes :sweat_smile: